iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -197,6 +197,47 @@ private async void ConnectionStatusChangeHandler(ConnectionStatus status, Connec
             }
         }
 
+        private async Task SubscribeToTwinUpdateNotificationsAsync(CancellationToken cancellationToken)
+        {
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                if (IsDeviceConnected)
+                {
+                    await s_deviceClient.SetDesiredPropertyUpdateCallbackAsync(HandleTwinUpdateNotificationsAsync, cancellationToken);
+                    break;
+                }
+
+                await Task.Delay(s_sleepDuration);
+            }
+        }
+
+        private async Task HandleTwinUpdateNotificationsAsync(TwinCollection twinUpdateRequest, object userContext)
+        {
+            CancellationToken cancellationToken = (CancellationToken)userContext;
+
+            if (!cancellationToken.IsCancellationRequested)
+            {
+                var reportedProperties = new TwinCollection();
+
+                _logger.LogInformation($"Twin property update requested: \n{twinUpdateRequest.ToJson()}");
+
+                // For the purpose of this sample, we'll blindly accept all twin property write requests.
+                foreach (KeyValuePair<string, object> desiredProperty in twinUpdateRequest)
+                {
+                    _logger.LogInformation($"Setting property {desiredProperty.Key} to {desiredProperty.Value}.");
+                    reportedProperties[desiredProperty.Key] = desiredProperty.Value;
+                }
+
+                await RetryOperationHelper.RetryTransientExceptionsAsync(
+                        async () =>
+                        {
+                            await s_deviceClient.UpdateReportedPropertiesAsync(reportedProperties, cancellationToken);

```

Comment:
Requires a comment that a device probably wouldn't blindly accept property updates and should do some checking, but for the sake of simplicity we do not do validation in this sample.

Reply:
Do you think moving the comment in L224 to the reported API call would make this clearer?

Reply:
Yes, best make it as contextually relevant to where the check would have happened.